### PR TITLE
Add a method to return the keyboard size. (Not including input accessory bar). 

### PR DESCRIPTION
**Pull request by jasperblues**

### CODE REVIEW

**Comment by stanislaw** [comment]
@jasperblues, I want to let you know that I've added this your `(CGSize)keyboardSize` method to my own collection of extensions for UIKit framework: [UIKitExtensions](https://github.com/stanislaw/UIKitExtensions) (I've modified your method slightly so it could fit the internals of my project). See `UIKitExtensions/UIScreen/Core.h`.

Also, I am ready to accept more reasonable methods to be added into the UIKitExtensions - feel free to suggest any if you like the way it is currently designed and written.

I really like `CKUITools` collection but I wanted to have a collection of this kind to be more generic and well organized. I would happily borrow some methods from `CKUITools` (with reference to the source, off course) - this will be another issue I will open soon.

Thanks.

**Comment by jasperblues** [comment]
@stanislaw - You might find Kevin Oneil's UsefulBits another good source  of ideas. . 

Regarding CKUITools - I think for its current size that its nicely organized, but we look forward to seeing what you come up with.

**Comment by stanislaw** [comment]
Thanks for response.

Thanks for mentioning UsefulBits - I didn't know about it. Really interesting collection. I've added it to Similar tools section and will explore its internals for sure))

> Regarding CKUITools - I think for its current size that its nicely organized, but we look forward to seeing what you come up with. 

Yes, size becomes a problem when a collection of this kind grows more and more. So while holding attention on this problem (btw Foundation is a huge number of methods many of which we use only once or never) I'm going to keep UIKitExtensions growing further, keeping it in accordance with the following principles:

**1)** Modularity
**2)** Careful naming (Category names, methods, variables) - encouraging Objective-C style, discouraging non-Objective-C.

**TLDR begins**

The most recent example is the UsefulBits library. Consider the following method from NSArray+Blocks:

``` objective-c
- (NSArray *)map:(id (^)(id item))block;
- (NSArray *)map:(id (^)(id item))block filterNil:(BOOL)filter_nil;
```

The both methods introduce non-Objective-C style: the first method looks like it would in Ruby (we have Array#map method there).  My FoundationExtensions (they are extensions-brothers with UIKitExtensions) has the same method, but it is called pretty much more like it would if it was a part of Foundation framework: 

``` objective-c
- (NSArray *)mapObjectsUsingBlock:(id(^)(id element, NSUInteger idx, BOOL *stop))block;
```

The second method `map:filterNil:` besides being non-Objective-C introduces additional noise having this part: `filterNil:(BOOL)filter_nil;` - we do see **filterNil** (Objective-C, OK) and **filter_nil** (OMG!) words staying alongside.

Also mine variant lives in `FoundationExtensions/NSArray/Mapping` pack which IMO is a much more precise and accurate naming for the purpose of this method. UsefulBits's NSArray+Blocks while being good in the functionality it implements - is a confused mixture of methods doing different tasks not related to each other. 

Foundation's NSArray itself does have a large amount of methods utilizing blocks functionality - so I don't think it deserves a separate category `+Blocks` to collect "anything that runs blocks"

**TLDR ends**

**3)** Following Objective-C syntax, not mixing in other languages style (Ruby is the most popular in this aspect - [ObjectiveSugar](https://github.com/mneorr/ObjectiveSugar), which I who have come from Ruby/Rails world especially don't like when I see it in Objective-C based projects IMO, IMO, IMO).

**4)** Reject methods which are stupid one-line helpers, do not reuse stuff that I use to meet only once in my daily routine.

**5)** Documentation. 

It is the problem of the most of such collections. CKUITools does not have any, UsefulBits does not have any... I can google dozens and more similar collections which look like "Hey, guys, see what I found useful here, look at these my *.{h,m} files". For me documentation is a holy grail and a pain in the same time - my ukrainian english is a challenge when it comes to "spell and compose paragraphs". 

---

Having that said, I want to excuse my appearance being (or looking) somehow impolite against CKUITools in my first comment above - I borrowed some of its methods (I described these in Credits section in README), I am subscribed to and review its issue notifications so I do really take your work into account. Let's keep in touch!

**Comment by genericspecific** [comment]
Stanislaw. What you are doing sounds great. Really useful and community minded. CKUITools was just a bunch of utilities that I used every day that a few people asked me to share. Nothing like what you have in mind. I'll be following your work closely...

Best,
Dino.

...sent from my phone...
